woocommerce/woocommerce-android
# Generate variations with bottom sheet picker

Closes: #7906
Closes: #7919

### Why
During the design review, we wanted to explore whether adding a bottom sheet would be a better experience for selecting between generating a new variant or generating all variants.

### Description
This PR adds a bottom sheet dialog for selecting between generating a new variant or generating all variants. 
Trying to generate all variants in a product that already has all the attributes combinations will result in an error. This PR also hides the generating all variants option from the bottom sheet dialog when the product already has all the attribute combinations variants.

### Testing instructions
1. Add a variable product with at least 1 attribute and no variations
2. Tap on the Add variation button within the Add your first variation Screen
3. Check that a bottom sheet dialog is opened with the options: Add new variation and Generate all variations
4. Check that generate all variations option shows the number of new variations to add
5. Tap on Add new variation
6. Check that a new variation is added
7. Press Back
8. Tap on Add Variation
9. Check that a bottom sheet dialog is opened with the options: Add new variation and Generate all variations
10. Check that generate all variations option shows the number of new variations to add.
11. Tap on generate all variations
12. Tap on OK
13. Check that the new variations are generated
14. Tap on Add variation
15. Check that now the bottom sheet only offers the option to Add a new variation


### Images/gif


https://user-images.githubusercontent.com/18119390/203624290-93c91cdc-c6dd-483c-8263-27193a86aab3.mp4
